src/box/lua/connpool.lua

```diff
@@ -118,11 +186,18 @@ function pool_methods.connect(self, instance_name, opts)
         end
         conn.mode = mode
         conn._deadline = old_deadline
+        conn._reconnect = old_reconnect
         local function watch_status(_key, value)
             conn._mode = value.is_ro and 'ro' or 'rw'
             self._connection_mode_update_cond:broadcast()
         end
         conn:watch('box.status', watch_status)
+        local function on_disconnect()
+            local reconnect_after = config:get('connpool.reconnect_after')
```

Comment:
BTW, `idle_timeout` is propagated to the connpool differently. Is there a reason to use two different approaches?